src/app/search/searchActions.js

```diff
@@ -61,43 +59,88 @@ const loadSearchResults = debounce((searchTerm, page, state, dispatch) => {
       type: SET_SEARCH_RESULTS,
       payload: {
         entries: [],
-        pageCount: 0,
       },
     })
     return
   }
 
-  const tilesets = searchableAndVisibleLayers.map((l) => l.tilesetId).join(',')
-  const baseURL = `${
-    process.env.REACT_APP_V2_API_ENDPOINT
-  }/tilesets/{{tilesets}}/search/?query={{query}}&limit={{limit}}&offset={{offset}}`
-  const url = buildEndpoint(baseURL, {
-    tilesets,
-    query: encodeURIComponent(searchTerm),
-    limit: SEARCH_MODAL_PAGE_SIZE,
-    offset: page * SEARCH_MODAL_PAGE_SIZE,
+  // NOW - error 500
+  // https://api-dot-skytruth-pleuston.appspot.com/v2/tilesets/gfw-tasks-657-uvi-v2,gfw-tasks-658-reefer-uvi-v4/search/?query=orca&limit=14&offset=0
+
+  // before - ok 200
+  // https://api-dot-skytruth-pelagos-production.appspot.com/v2/tilesets/gfw-tasks-657-uvi-v2,test-chile-seconds-aquaculture-v7,test-chile-seconds-transport-v7,test-chile-seconds-industry-v7,test-chile-seconds-small-fisheries-v7,gfw-tasks-658-reefer-uvi-v4,/search/?query=orca&limit=14&offset=0
+  const searchEndpoints = searchableAndVisibleLayers
+    .map((l) => {
+      let endpoint = {
+        url: l.header.endpoints.search,
+        tilesetId: l.tilesetId,
+      }
+      // TODO remove this hack
+      endpoint.url = endpoint.url.replace('gfw-tasks-657-uvi-v2', '{{tilesets}}')
+      endpoint.url = endpoint.url.replace('gfw-tasks-658-reefer-uvi-v4', '{{tilesets}}')
+      endpoint.url = endpoint.url.replace('gfw-task-673-encounters-v3', '{{tilesets}}')
+      // endpoint.url = endpoint.url.replace('api-dot-skytruth-pleuston.appspot.com', 'api-dot-skytruth-pelagos-production.appspot.com')
+      return endpoint
+    })
+    .filter((l) => l.tilesetId !== undefined)
+
+  // Deduplicate search endpoints. Keep all layer tilesetIds to build the final URL
+  const uniqSearchEndpoints = {}
+  searchEndpoints.forEach((searchEndpoint) => {
+    if (uniqSearchEndpoints[searchEndpoint.url] === undefined) {
+      uniqSearchEndpoints[searchEndpoint.url] = []
+    }
+    uniqSearchEndpoints[searchEndpoint.url].push(searchEndpoint.tilesetId)
   })
 
-  fetch(url, options)
-    .then((response) => response.json())
-    .then((result) => {
-      if (result.query !== searchTerm) {
-        console.warn('search term differs, searching for', searchTerm, 'receiving', result.query)
+  // build uniq search URLs with all get params
+  const uniqSearchURLs = Object.keys(uniqSearchEndpoints).map((searchEndpointURL) => {
+    return buildEndpoint(searchEndpointURL, {
+      tilesets: uniqSearchEndpoints[searchEndpointURL].join(','),
+      query: encodeURIComponent(searchTerm),
+      limit: SEARCH_MODAL_PAGE_SIZE,
+      offset: page * SEARCH_MODAL_PAGE_SIZE,
+    })
+  })
+
+  const searchPromises = uniqSearchURLs.map((url) =>
+    fetch(url, options)
+      .then((response) => response.json())
+      .catch((err) => err)
+  )
+
+  Promise.all(searchPromises).then((resultsOrErrors) => {
```

Comment:
I don't think this will return an array of all searchs with an error, following this: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all#Rejection because if any of the promises fail it won't go go the `then` method but `catch` instead. Have you checked ?

Reply:
Yes. When one of the promises fail, it will go to the Promise.all `then`. Not sure why.